rfc/languages-in-w3s.md

```diff
@@ -0,0 +1,84 @@
+- Go's ability to do binary distributions by default is a major plus for kind of project.
+- By and large, people get stuff done quickly in Go because Go is extremely simple. It's very fast to learn, easy to read, and it all looks the same. The only problem is its extremely boring. (Note from Hannah: When I first started writing Go I hated it. I thought someone took the joy out of programming. Now I love it cause I get my joy from what I'm building, not my programming language)
+- Lots and lots of Filecoin tooling and software is written in Go. In fact, W3S already has Go in it, we just don't maintain it -- it's called Spade
+- Garbage collection and green threads present challenges for using Go over an FFI, or for compiling go to WebAssembly, though TinyGo has improved this story by reducing the runtime size.
+
+### Rust
+
+Caveat from Hannah: I am not a highly experienced Rust dev
+
+- Rust is a better C/C++. Like definitely much better. This is huge cause no one wrote a better C/C++ for low level programming for 20 years.
+- However, by and large, you should probably not write programs in Rust unless you'd consider writing them in C/C++. There is minimal evidence one can be as productive in Rust as higher level languages.
+- In particular, you should probably not write web servers in Rust or any other asynchronous programs unless you have extra time on your hands which we don't.
+- BTW, if you haven't programmed in C/C++ recently, you've probably forgotten that long compile times are a thing on large projects in low level languages. 
+- Rust is essential for highly performant synchronous compute tasks. That's why people use it for smart contracts, hashing, and proofs.
+- Rust has the best WebAssembly story of any language.
+
+## Table Stakes For Considering A Language for W3S
+
+In addition to having a compelling engineering or business reason, we have some other things to consider before we adopt a language:
+
+- UCAN libraries and tooling especially for anything with a web interface
+- IPLD/IPFS/Libp2p libraries (we may not always need libp2p going forward)
+
+## Proposing a language to adopt
+
+- The best time to propose a language to adopt is at the start of a new project
+- Present a proposal to write a project in a new language via an RFC
+   - Try to objectively present business and engineering tradeoffs of writing the project in a language we already support vs writing it in a new language
+   - Identify gaps in language support or tooling related to our software stack and present a plan to address it
+   - Ideally, make a case for why this will be useful for other projects going forward
+   - Present a plan for bringing other team members up to speed on the language you propose to use. Usually the language proposer will have to captain the process of language education.
+- Since it's a big decision, we should try to get input from all devs before making a decision at least till our team gets bigger
```

Comment:
> Since it's a big decision

Hopefully, it will not be a big decision, because the engineering requirements and design will naturally select a set of tools that best enables the creation, deployment, and maintenance of the designed solution.